src/test/java/org/javarosa/xpath/test/XPathEvalTest.java

```diff
@@ -99,6 +99,24 @@ public void testTypeCoercion(){
     }
 
     @Test
+    public void sortTests() {
+        // simple sort
+        testEval("sort('commcare is the best tool ever', false())", null, null, "tool the is ever commcare best");
+
+        // sort 2nd list by 1st
+        testEval("sort('2222 5555 9999 1111', 'd b c a', true())", null, null, "1111 5555 9999 2222");
+        testEval("sort('a b c d e f', '4 2 1 5 3 2', false())", null, null, "d a e f b c");
+        testEval("sort('c c z f z f', '4 2 1 5 3 2', true())", null, null, "z c f z c f");
```

Comment:
Hey @ctsims sorry I'm a little confused, the right side lists of a lot of these tests already do have repeats. `'4 2 1 5 3 2'` has 2 in it twice, and the way it handles deciding on the order for the items on the left side which both match to `2` is to sort by those items themselves in the direction of the sort. So that's why:

in `sort('a b c d e f', '4 2 1 5 3 2')`, `b` comes before `f` in the output  ( `c b f e a d`)
but 
in `sort('a b c d e f', '4 2 1 5 3 2', false())`, `f` comes before `b` in the output ( `d a e f b c`)